### PR TITLE
Introduce javadoc and scaladoc directives for cluster-dc.md

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #22904

paradox pages generated in my firebase demo site:
https://richard-akka-doc.firebaseapp.com/cluster-dc.html

Confirmed hyperlinks work fine.

Although lines are getting very long horizontally, I am not adding line breaks as that would make reviewing the diff very difficult, especially if I am adding new commits from here.
Once everything looks ok and we still want line breaks, I can add them in the end.

### CODE REVIEW

File: akka-docs/src/main/paradox/cluster-dc.md [hunk 1]
```diff
@@ -112,11 +112,11 @@ Two different failure detectors can be configured for these two purposes:
 * `akka.cluster.failure-detector` for failure detection within own data center
 * `akka.cluster.multi-data-center.failure-detector` for failure detection across different data centers
 
-When @ref[subscribing to cluster events](cluster-usage.md#cluster-subscriber) the `UnreachableMember` and
-`ReachableMember` events are for observations within the own data center. The same data center as where the
+When @ref[subscribing to cluster events](cluster-usage.md#cluster-subscriber) the @scala[@scaladoc[`UnreachableMember`](akka.cluster.ClusterEvent$)]@java[`UnreachableMember`] and
+@scala[@scaladoc[`ReachableMember`](akka.cluster.ClusterEvent$)]@java[@javadoc[`ReachableMember`](akka.cluster.ClusterEvent$)] events are for observations within the own data center. The same data center as where the
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/cluster-dc.md]: ### scala
scaladoc is linked to:
https://doc.akka.io/api/akka/2.5/akka/cluster/ClusterEvent$.html

As the URL for `UnreachableMember` includes the return type, which is messy.
https://doc.akka.io/api/akka/2.5/akka/cluster/ClusterEvent$.html#UnreachableMemberextendsClusterEvent.ReachabilityEventwithProductwithSerializable

### java
javadoc link is not added as there is no way to get it work. 
The correct URL is:
https://doc.akka.io/japi/akka/2.5/akka/cluster/ClusterEvent.UnreachableMember.html

but `@javadoc[...](akka.cluster.ClusterEvent.UnreachableMember)`  will be linked to 
https://doc.akka.io/japi/akka/2.5/akka/cluster/ClusterEvent/UnreachableMember.html (`/` after ClusterEvent) 
which doesn't exist.

File: akka-docs/src/main/paradox/cluster-dc.md [hunk 2]
```diff
@@ -158,8 +158,8 @@ Scala
 Java
 :  @@snip [ClusterSingletonManagerTest.java]($akka$/akka-cluster-tools/src/test/java/akka/cluster/singleton/ClusterSingletonManagerTest.java) { #create-singleton-proxy-dc }
 
-If using the own data center as the `withDataCenter` parameter that would be a proxy for the singleton in the own data center, which
-is also the default if `withDataCenter` is not given.
+If using the own data center as the @scala[@scaladoc[`withDataCenter`](akka.cluster.singleton.ClusterSingletonProxySettings)]@java[@javadoc[`withDataCenter`](akka.cluster.singleton.ClusterSingletonProxySettings)] parameter that would be a proxy for the singleton in the own data center, which
+is also the default if @scala[@scaladoc[`withDataCenter`](akka.cluster.singleton.ClusterSingletonProxySettings)]@java[@javadoc[`withDataCenter`](akka.cluster.singleton.ClusterSingletonProxySettings)] is not given.
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/cluster-dc.md]: Similar to [this comment](https://github.com/akka/akka/pull/24525/files#r167432330), scaladoc to a method includes return type, which is messy. Similar reason for javadoc.

**Comment by richardimaoka** [comment]
Seems like a transient error. https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/11719/console

```
[error] (whitesourceCheckPolicies) Error communicating with service: Server error: java.lang.NullPointerException
[error] Total time: 13 s, completed Feb 11, 2018 1:24:32 PM
```

Ready for review.

File: akka-docs/src/main/paradox/cluster-dc.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/cluster-dc.md]: And the class name from Java's point of view does not work either? I think that'd be
`akka.cluster.ClusterEvent$.UnreachableMember`

**Comment by richardimaoka** [comment]
Superseded by #24689

**Pull request closed without merging**